### PR TITLE
feat: Addition DX arguments

### PR DESCRIPTION
**Pull request by pstreef**

### CODE REVIEW

File: administrator-documentation/moderne-dx/how-to-guides/dx-variables.md [hunk 1]
```diff
@@ -188,7 +188,9 @@ java -jar moderne-dx-{version}.jar \
 {% tab title="OCI Container" %}
 **Variables:**
 
-* `MODERNE_DX_ORGANIZATION_URL` – _The URL of your GraphQL service that provides organization information._
+* `MODERNE_DX_ORGANIZATION_URL` – _The URL of your GraphQL service that provides organization information. Cannot be combined with `MODERNE_DX_ORGANIZATION_REPOSCSV`_
+* `MODERNE_DX_ORGANIZATION_REPOSCSV` - _The path of your repos.csv file that provides organization information. Cannot be combined with `MODERNE_DX_ORGANIZATION_URL`_
```
Comment (kmccarp) [inline administrator-documentation/moderne-dx/how-to-guides/dx-variables.md:192]: Should we say which one will take precedence, or do we prevent it from starting up?

Comment (pstreef) [reply administrator-documentation/moderne-dx/how-to-guides/dx-variables.md:192]: Startup fails if not exactly 1 is set (if any org configuration exists)

**Comment by okundzich** [comment]
what is this about? it has no ticket or description provided

**Comment by mike-solomon** [comment]
Thanks for keeping the docs up-to-date! Would you mind updating both the all variables doc and [the individual DX org var doc](https://github.com/moderneinc/moderne-docs/blob/main/administrator-documentation/moderne-dx/how-to-guides/configure-dx-org-service.md) please?

**Comment by pstreef** [comment]
> what is this about? it has no ticket or description provided

Since this is a change to documentation I assumed that description is a bit redundant.

here is the slack thread: https://moderneinc.slack.com/archives/C01S4M53THT/p1721677494055959

**Comment by pstreef** [comment]
> Thanks for keeping the docs up-to-date! Would you mind updating both the all variables doc and [the individual DX org var doc](https://github.com/moderneinc/moderne-docs/blob/main/administrator-documentation/moderne-dx/how-to-guides/configure-dx-org-service.md) please?
> 
> The doc you're updating is just the list of vars for reference - but we have individual docs that go into more details about the vars.

using `moderne.dx.organization.reposCsv` is an alternative to running the org-service. Maybe this should have a separate document? 🤔

**Comment by mike-solomon** [comment]
> using `moderne.dx.organization.reposCsv` is an alternative to running the org-service. Maybe this should have a separate document? 🤔

They look like they're tied together in a lot of ways. Like you either have to use one or the other from the descriptions. I think it would make sense to just update the organization service vars doc and potentially add some context for that. Otherwise, it might be even more confusing swapping between the docs.

**Review by mike-solomon — APPROVED** [review]
Changes look good to me! Cheers :D

**Comment by pstreef** [comment]
> They look like they're tied together in a lot of ways. Like you either have to use one or the other from the descriptions. I think it would make sense to just update the organization service vars doc and potentially add some context for that. Otherwise, it might be even more confusing swapping between the docs.

hmm, good point. I think renaming it from "Configure the Organizations service with Moderne DX" to "Configure the Organizations data source with Moderne DX" would be a good start.

I will look into that

**Comment by pstreef** [comment]
> Changes look good to me! Cheers :D

I did not add the repos.csv part to the other docs yet.

**Comment by mike-solomon** [comment]
> > Changes look good to me! Cheers :D
> 
> I did not add the repos.csv part to the other docs yet.

No worries! I figured there may still be additional changes - but wanted to stamp so you're not blocked. Docs are easy enough to update after the fact if we notice any issues ^.^

**Comment by pstreef** [comment]
@mike-solomon I made significant changes, please re-review!

File: administrator-documentation/moderne-dx/how-to-guides/configure-dx-organizations.md [hunk 2]
```diff
@@ -12,14 +12,13 @@ This guide assumes that:
...
 
-You have two main options for building this service. You can:
-
-1. (**Recommended**) Fork our [Organizations service template](https://github.com/moderneinc/moderne-organizations) and modify it to meet your needs. Please see the [README](https://github.com/moderneinc/moderne-organizations/blob/main/README.md) for how to spin this up quickly. It can be as simple as updating a CSV file.
-2. Build your own service that fulfills the [GraphQL contract](https://github.com/moderneinc/moderne-organizations/blob/main/src/main/resources/schema/moderne-organizations.graphqls) using any GraphQL stack (e.g., NodeJS, Rust, C#, etc.)
+The simplest way to achieve the organization structure is to supply a repos.csv file directly in DX.
 
-We generally recommend forking the template and modifying it as, in most cases, that will be faster and easier than building it yourself. Regardless of which one you choose, however, some developer time will be required on your end.
+Downsides of file based organization structure:
+- You cannot configure DevCenter
```
Comment (mike-solomon) [inline administrator-documentation/moderne-dx/how-to-guides/configure-dx-organizations.md:20]: I struggled to parse this a bit at first as I thought there were multiple DevCenters. I think we could fix this by just slightly rephrasing this and adding a link:

```suggestion
- You cannot configure any [DevCenter](/administrator-documentation/moderne-platform/how-to-guides/dev-center.md)
```

File: administrator-documentation/moderne-dx/how-to-guides/configure-dx-organizations.md [hunk 3]
```diff
@@ -12,14 +12,13 @@ This guide assumes that:
...
 * You have already [configured Moderne DX to connect to your Artifactory instance](configure-dx-with-artifactory-access.md)
 
-## Organizations service template and API
+## File based organization structure
 
-You have two main options for building this service. You can:
-
-1. (**Recommended**) Fork our [Organizations service template](https://github.com/moderneinc/moderne-organizations) and modify it to meet your needs. Please see the [README](https://github.com/moderneinc/moderne-organizations/blob/main/README.md) for how to spin this up quickly. It can be as simple as updating a CSV file.
-2. Build your own service that fulfills the [GraphQL contract](https://github.com/moderneinc/moderne-organizations/blob/main/src/main/resources/schema/moderne-organizations.graphqls) using any GraphQL stack (e.g., NodeJS, Rust, C#, etc.)
+The simplest way to achieve the organization structure is to supply a repos.csv file directly in DX.
```
Comment (mike-solomon) [inline administrator-documentation/moderne-dx/how-to-guides/configure-dx-organizations.md:17]:
```suggestion
The simplest way to achieve the organization structure is to supply a `repos.csv` file directly in DX.
```

File: administrator-documentation/moderne-dx/how-to-guides/configure-dx-organizations.md [hunk 4]
```diff
@@ -12,14 +12,13 @@ This guide assumes that:
...
 
-We generally recommend forking the template and modifying it as, in most cases, that will be faster and easier than building it yourself. Regardless of which one you choose, however, some developer time will be required on your end.
+Downsides of file based organization structure:
```
Comment (mike-solomon) [inline administrator-documentation/moderne-dx/how-to-guides/configure-dx-organizations.md:19]:
```suggestion
That being said, there are a few downsides of file based organization structure:
```

File: administrator-documentation/moderne-dx/how-to-guides/configure-dx-organizations.md [hunk 5]
```diff
@@ -39,25 +38,39 @@ The above example would be used in Moderne DX to generate an organizational list
 
 To generate this `repos.csv`, we recommend using "[repo fetchers](https://github.com/moderneinc/moderne-organizations/blob/main/repo-fetchers/README.md)" inside our Organizations service template.
 
-## Deploying the service
+## Service based organization structure
+
+A more feature rich replacement of the file based approach is to run the Organizations service. This allowes you to use additional features like the DevCenter, organization based access control and per organization commit options.
```
Comment (mike-solomon) [inline administrator-documentation/moderne-dx/how-to-guides/configure-dx-organizations.md:43]:
```suggestion
A more feature rich replacement of the file based approach is to create an [Organizations service](/administrator-documentation/moderne-dx/how-to-guides/configure-dx-org-service.md). This allows you to use additional features like the DevCenter, organization based access control, and per organization commit options.
```

File: administrator-documentation/moderne-dx/how-to-guides/configure-dx-organizations.md [hunk 6]
```diff
@@ -39,25 +38,39 @@ The above example would be used in Moderne DX to generate an organizational list
...
+
+### Deploying the service
 
 How you deploy the service is largely up to your company. With that being said, there are a few important things to be aware of:
 
-* Communication with the Organizations service is done through the [Moderne agent](../../moderne-platform/how-to-guides/agent-configuration/agent-configuration.md). Therefore, this service **must** be accessible from the agent.
-* Moderne will make a request per repository to the Organizations service once every 10 minutes by default (you can change this interval in your [agent configuration](../../moderne-platform/how-to-guides/agent-configuration/configure-organizations-service.md)). Please ensure that you have metrics to track how this service is performing so you can adjust it over time.
+* Moderne will make a request per repository to the Organizations service once every 10 minutes by default (you can change this interval in your configuration. Please ensure that you have metrics to track how this service is performing so you can adjust it over time.
```
Comment (mike-solomon) [inline administrator-documentation/moderne-dx/how-to-guides/configure-dx-organizations.md:58]:
```suggestion
* Moderne will make a request per repository to the Organizations service once every 10 minutes by default (you can change this interval in your configuration). Please ensure that you have metrics to track how this service is performing so you can adjust it over time.
```

File: administrator-documentation/moderne-dx/how-to-guides/dx-configuration.md [hunk 7]
```diff
@@ -249,11 +249,11 @@ java -jar moderne-dx-{version}.jar \
 {% endtab %}
 {% endtabs %}
 
-### Step 4: (Optionally) Configure the Organizations service
+### Step 4: (Optionally) Configure an organization structure
 
-Many organizations desire the ability to control the organizational structure of their repositories within the Moderne Platform in a dynamic way. To facilitate this need, Moderne provides an optional integration with an Organizations service that is hosted inside of your environment.
+Many organizations desire the ability to control the organizational structure of their repositories within the Moderne Platform in a dynamic way. To facilitate this need, Moderne provides two approaches, either a file based one or by integration with an Organizations service that is hosted inside of your environment.
```
Comment (mike-solomon) [inline administrator-documentation/moderne-dx/how-to-guides/dx-configuration.md:254]:
```suggestion
Many organizations desire the ability to control the organizational structure of their repositories within the Moderne Platform in a dynamic way. To facilitate this need, Moderne provides two approaches: a file-based one or a service-based one (where you configure an Organizations service that is hosted inside of your environment).
```

File: administrator-documentation/moderne-dx/how-to-guides/dx-configuration.md [hunk 8]
```diff
@@ -249,11 +249,11 @@ java -jar moderne-dx-{version}.jar \
...
 
-If you want to set up this service, please see the [organizations service setup instructions](../../moderne-platform/how-to-guides/organizations-service.md). Then, once it has been set up, please [configure the Moderne DX service to connect with the organization service](configure-dx-org-service.md).
+If you want to set up this integration, please see [configure Organizations with Moderne DX](configure-dx-organizations.md).
```
Comment (mike-solomon) [inline administrator-documentation/moderne-dx/how-to-guides/dx-configuration.md:256]:
```suggestion
If you want to set up this integration, please [follow tine instructions in our configuring Organizations with Moderne DX doc](configure-dx-organizations.md).
```

**Review by mike-solomon — APPROVED** [review]
A few minor suggestions - but other than that looks good to me :D - thanks so much!

File: SUMMARY.md [hunk 9]
```diff
@@ -110,7 +110,7 @@
     * [Configure Moderne DX with Maven repository access](administrator-documentation/moderne-dx/how-to-guides/configure-dx-with-maven-repository-access.md)
     * [Configure Moderne DX with Artifactory access: LSTs](administrator-documentation/moderne-dx/how-to-guides/configure-dx-with-artifactory-access.md)
     * [Configure Moderne DX with Artifactory access: recipes](administrator-documentation/moderne-dx/how-to-guides/configure-dx-with-artifactory-recipes.md)
-    * [Configure the Organizations service with Moderne DX](administrator-documentation/moderne-dx/how-to-guides/configure-dx-org-service.md)
+    * [Configure Organizations with Moderne DX](administrator-documentation/moderne-dx/how-to-guides/configure-dx-organizations.md)
```
[the remark this answers is not in the document]
Comment (pstreef) [reply SUMMARY.md:113]: I changed it in all other places, but external links will indeed break..

**Pull request merged**